src/LanguageServerProtocol/LanguageServerProtocol.fs

```diff
@@ -2399,36 +2445,111 @@ module Server =
             }
             loop ())
 
-        /// When the server wants to send a request/notification to the client
-        let sendServerRequest (rpcMethod: string) (requestObj: obj): AsyncLspResult<unit> =
-            let serializedResponse = JToken.FromObject(requestObj, jsonSerializer)
-            let req = JsonRpc.Request.Create(rpcMethod, serializedResponse)
-            let reqString = JsonConvert.SerializeObject(req, jsonSettings)
-            sender.Post(reqString)
-            // TODO: Really wait for the client answer if not a notification (Necessary to implement requests)
+        let responseAgent = MailboxProcessor<ResponseMailboxMsg>.Start(fun agent ->
```

Comment:
I like having the send and receive agents on different processors :thumbsup: that should help reduce any throughput issues.